main/discovery_beacon.c

```diff
@@ -0,0 +1,118 @@
+#include "discovery_beacon.h"
+
+#include "driver_setup.h"
+#include "esp_log.h"
+#include "lwip/sockets.h"
+// #include "esp_netif.h"
+
+// TODO: Finish this up!!
+
+// Name that will be used for logging
+static const char* TAG = "discovery_beacon";
+
+// Static memory for the discovery_beacon_task stack.
+static StackType_t task_stack[4096];
+
+// Static memory for the discovery_beacon_task.
+static StaticTask_t task_mem;
+
+static void discovery_beacon_task(void* pvParameters);
+
+int discovery_beacon_start() {
+  // create a UDP socket
+  int server_sock = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+  if (server_sock < 0) {
+    ESP_LOGE(TAG, "Unable to create socket: errno %d", errno);
+    return -1;
+  }
+
+  // Enable broadcast functionality
+  int broadcastEnable = 1;
+  int err = setsockopt(server_sock, SOL_SOCKET, SO_BROADCAST, &broadcastEnable,
+                       sizeof(broadcastEnable));
+  if (err != 0) {
+    ESP_LOGE(TAG, "Unable to enable UDP broadcast on socket: %d", errno);
+    close(server_sock);
+    return -1;
+  }
+
+  xTaskCreateStatic(discovery_beacon_task, "discovery_beacon",
+                    sizeof(task_stack), (void*)server_sock, 0, task_stack,
+                    &task_mem);
+  return 0;
+}
+
+static void discovery_beacon_task(void* pvParameters) {
+  int server_sock = (int)pvParameters;
+
+  // Set up the broadcast address
+  struct sockaddr_in broadcast_addr = {0};
+  broadcast_addr.sin_addr.s_addr = inet_addr("255.255.255.255");
+  broadcast_addr.sin_family = AF_INET;
+  broadcast_addr.sin_port = htons(42000);
+
+  char msg_buf[1024];
```

Comment:
Allocating larger buffers on the stack should be avoided. I'd recommend statically allocating this (preferred) or using malloc since we have dynamic memory allocation.

Reply:
Fixed in 72219c3 by moving this buffer to static memory.